contracts/src/GMXSampler.sol

```diff
@@ -21,13 +21,22 @@ contract GMXSampler is SamplerUtils, ApproximateBuys {
         uint256 numSamples = takerTokenAmounts.length;
         makerTokenAmounts = new uint256[](numSamples);
         for (uint256 i = 0; i < numSamples; i++) {
-            try IGMX(reader).getAmountOut(IVault(vault), path[0], path[1], takerTokenAmounts[i]) returns (
-                uint256 amountAfterFees,
-                uint256 feeAmount
-            ) {
-                makerTokenAmounts[i] = amountAfterFees;
-                // Break early if there are 0 amounts
-                if (makerTokenAmounts[i] == 0) {
+            try IGMX(reader).getMaxAmountIn(IVault(vault), path[0], path[1]) returns (uint256 maxAmountIn) {
```

Comment:
could we do this once outside of the loop?

Reply:
yeah good point, didnt think about sampling happening within the same block so that value shouldnt change